### ISSUE TITLE
Send transaction fail: We do not appear to fully agree with our peers

### ISSUE THREAD

**Issue by jtomtan** [bitcoin-sv/bitcoin-sv#146]
version: bitcoinsv-1.0.1

when sending transaction: 

com.googlecode.jsonrpc4j.HttpException: {"result":null,"error":{"code":-2,"message":"Safe mode: Warning: We do not appear to fully agree with our peers! You may need to upgrade, or other nodes may need to upgrade."},"id":"-4701277518029195875"}

**Comment by 1scrooge** [comment]
The same issue. My version node 1.0.1 too.
```
"version": 101000100,
  "protocolversion": 70015,
  "walletversion": 160300,
  "balance": 0.00000000,
  "blocks": 620998,
  "timeoffset": 0,
  "connections": 8,
  "proxy": "",
  "difficulty": 393476117941.0039,
  "testnet": false,
  "stn": false,
  "keypoololdest": 1580883178,
  "keypoolsize": 2000,
  "paytxfee": 0.00000000,
  "relayfee": 0.00000250,
  "errors": "Warning: We do not appear to fully agree with our peers! You may need to upgrade, or other nodes may need to upgrade.",
  "maxblocksize": 2000000000,
  "maxminedblocksize": 128000000,
  "maxstackmemoryusagepolicy": 100000000,
  "maxstackmemoryusageconsensus": 200000000
```

How can I fix it?

**Comment by Danconnolly** [comment]
You're not running a BCH or BTC node from the same ip address are you?

**Comment by jtomtan** [comment]
> You're not running a BCH or BTC node from the same ip address are you?

No, only running BSV, even tried several other machines and several different of configurations.

When you restart the bitcoin-sv, it tries to connect to peers. At this time, send transaction is successful. As long as it finishes connection, A popup message is showing: 

2020-02-08 05:41:21 [bitcoin-worker4-N-CAsyncTaskPool] UpdateTip: new best=000000000000000000f48485d580b91b2492d0dd1e24d79793929b54f1150db7 height=621136 version=0x20000000 log2_work=87.959125 tx=350956949 date='2020-02-08 05:40:55' progress=1.000000 cache=25.9MiB(155678txo) warning='58 of last 100 blocks have unexpected version'
2020-02-08 05:41:21 [bitcoin-worker4-N-CAsyncTaskPool] CheckForkWarningConditions: Warning: Found invalid chain at least ~6 blocks longer than our best chain.
Chain state database corruption likely.

After this, try send transaction, always fail with the error message described on the first post.

**Comment by bitcartel** [comment]
@jtomtan Try one of the following commands, to invalidate the BCH chaintips (headers-only longer chain)

bitcoin-cli invalidateblock 0000000000000000008e8d83cba6d45a9314bc2ef4538d4e0577c6bed8593536

or

bitcoind -invalidateblock 0000000000000000008e8d83cba6d45a9314bc2ef4538d4e0577c6bed8593536

where the blockhash is block 600000 on BCH.

**Comment by jtomtan** [comment]
> @jtomtan Try one of the following commands, to invalidate the BCH chaintips (headers-only longer chain)
> 
> bitcoin-cli invalidateblock 0000000000000000008e8d83cba6d45a9314bc2ef4538d4e0577c6bed8593536
> 
> or
> 
> bitcoind -invalidateblock 0000000000000000008e8d83cba6d45a9314bc2ef4538d4e0577c6bed8593536
> 
> where the blockhash is block 600000 on BCH.

Doesn't work. The problems is related to the hardfork this time, I think it should not be related to BCH. I think the reason is that too many nodes do not upgrade to the newest version. 

I am very puzzled how did others send transaction after the hardfork. I think run the newest software should make everything working. But it doesn't. Even i run the bitcoin-sv 1.0.1 from a clean machine.

**Comment by 1scrooge** [comment]
> You're not running a BCH or BTC node from the same ip address are you?

I have BTC an BCH on the same external IP, but before updating SV node I had no problem. BCH & BTC works fine

**Comment by Danconnolly** [comment]
This will be fixed in the next version which we expect early next week.

In the meantime, we advise you use `disablesafemode=1` in the bitcoin.conf file.

**Comment by jtomtan** [comment]
> This will be fixed in the next version which we expect early next week.
> 
> In the meantime, we advise you use `disablesafemode=1` in the bitcoin.conf file.
disablesafemode=1 helps. Send transaction is working after this parameter is added.

**Comment by jtomtan** [comment]
> This will be fixed in the next version which we expect early next week.
> 
> In the meantime, we advise you use `disablesafemode=1` in the bitcoin.conf file.

disablesafemode=1 helps. Send transaction is working after this parameter is added. Thanks.
